README
NAME

    Dist::Zilla::PluginBundle::Author::CCM - A plugin bundle for
    distributions built by CCM

VERSION

    version 0.012

SYNOPSIS

        # In your dist.ini file:
        [@Author::CCM]

DESCRIPTION

    You probably don't want to use this.

        ; VERSION
        [Git::NextVersion]
    
        ; GATHER
        [Git::GatherDir]
        exclude_filename    = LICENSE
        exclude_filename    = README.md
        [PruneCruft]
        [PruneFiles]
        filename            = dist.ini
    
        [CopyFilesFromBuild]
        copy                = LICENSE
        [ExecDir]
    
        ; PREREQS
        [AutoPrereqs]
        [Prereqs::FromCPANfile]     ; if a cpanfile exists in root
        [Prereqs::AuthorDeps]
    
        ; TESTS
        [MetaTests]
        [Test::CPAN::Changes]
        [PodCoverageTests]
        [PodSyntaxTests]
        [Test::Pod::No404s]
        [Test::Compile]
        [Test::MinimumVersion]
        max_target_perl     = 5.10.1
        [Test::EOL]
        [Test::NoTabs]
        [Test::Perl::Critic]
        [Test::Portability]
        [Test::CleanNamespaces]
        [Test::ReportPrereqs]
    
        ; METADATA
        [Authority]
        do_munging          = 0
        [MetaJSON]
        [MetaYAML]
        [MetaNoIndex]
        directory           = eg
        directory           = share
        directory           = shares
        directory           = t
        directory           = xt
        [MetaProvides::Package]
        [Keywords]
        [Git::Contributors]
        order_by            = commits
        [GithubMeta]
        issues              = 1
    
        ; MUNGE
        [PodWeaver]
        config_plugin       = @Author::CCM
        [OverridePkgVersion]
    
        ; GENERATE FILES
        [License]
        [ReadmeAnyFromPod]
        filename            = README.md
        locaton             = root
        type                = markdown
        phase               = release
        [ReadmeAnyFromPod]
        filename            = README
        location            = build
        type                = text
        [Manifest]
        [ManifestSkip]
    
        [MakeMaker]                 ; override with the "installer" attribute
    
        ; RELEASE
        [NextRelease]
        [CheckChangesHasContent]
        [Git::Check]
        [RunExtraTests]
        [TestRelease]
        [ConfirmRelease]
        [UploadToCPAN]              ; disable with the "no_upload" attribute
        [Git::Commit]
        commit_msg          = Release %N %v%t%n%n%c
        [Git::CommitBuild]
        branch              =
        release_branch      = dist
        release_message     = Version %v%t
        [Git::Tag]
        tag_message         = Version %v%t%n%n%c
        [Git::Push]
        push_to             = origin master +master:refs/heads/release +dist
        remotes_must_exist  = 0

ATTRIBUTES

 max_target_perl

    Specify the minimum perl version. Defaults to 5.10.1.

 no_index

    Set directories to not index.

    Default:

 authority

    Specify the release authority. Defaults to cpan:CCM.

 installer

    Specify which installer to use, such as:

      * MakeMaker (default)

      * MakeMaker::Custom

 airplane

    Disable plugins that use the network, and prevent releasing.

 no_upload

    Do not upload to CPAN or git push.

METHODS

 configure

    Required by Dist::Zilla::Role::PluginBundle::Easy.

SEE ALSO

      * Dist::Zilla

      * Dist::Zilla::PluginBundle::Author::ETHER

BUGS

    Please report any bugs or feature requests on the bugtracker website
    https://github.com/chazmcgarvey/Dist-Zilla-PluginBundle-Author-CCM/issues

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR

    Charles McGarvey <ccm@cpan.org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2016 by Charles McGarvey.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Dist-Zilla-PluginBundle-Author-CCM.

0.012     2022-04-08 15:53:36-0600

  * Allow no installer

0.011     2022-03-31 08:24:44-0600

  * Change NextRelease datetime format

0.010     2022-03-29 20:09:03-0600

  * Add no_index option so MetaNoIndex is easily configurable

0.009     2018-11-22 22:46:42-0700

  * Remove use of DZP::TravisYML and DZP::ReversionOnRelease

0.008001  2017-08-25 22:00:56-0600

  * Declare new dependency on Perl::Version

0.008     2017-08-25 21:28:07-0600

  * Automatically figure out perl_version and perl_version_build TravisYML
    configuration based on configured minimum perl requirement

0.007     2017-03-11 18:33:58-0700

  * Prevent writing wrong version to Changes

0.006     2016-11-28 02:08:07-0700

  * Work around a bug in DZP::TravisYML

0.005     2016-11-27 19:52:27-0700

  * Move dependencies from cpanfile to dist.ini (fixes testing)
  * Set copyright year

0.004     2016-11-27 18:28:43-0700

  * This bundle now requires at least perl 5.14.0
  * Add Test::MinimumVersion as a release test

0.003     2016-11-26 00:09:53-0700

  * Set default authority to cpan:CCM
  * Add Test::CPAN::Meta to prereqs

0.002     2016-11-25 22:12:27-0700

  * Add DZIL_NO_UPLOAD to enable no_upload mode
  * Only rewrite repo README on release
  * Use RunExtraTests instead of ExtraTests
  * Use Bootstrap::lib
  * Require Dist::Zilla 5.038 (for :ExtraTestFiles)

0.001     2016-11-25 18:28:13-0700

  * Initial version



lib/Pod/Weaver/PluginBundle/Author/CCM.pm
package Pod::Weaver::PluginBundle::Author::CCM;
# ABSTRACT: A plugin bundle for pod woven by CCM


use 5.008;
use warnings;
use strict;

our $VERSION = '0.012'; # VERSION

use Pod::Weaver::Config::Assembler;
use namespace::autoclean;


sub configure {
    return (
        ['-EnsurePod5'],
        ['-H1Nester'],
        ['-SingleEncoding'],

        ['-Transformer' => List     => {transformer => 'List'}],
        ['-Transformer' => Verbatim => {transformer => 'Verbatim'}],

        ['Region' => 'header'],

        'Name',
        # ['Badges' => {badge => [qw(perl travis coverage)], formats => 'html, markdown'}],

        'Version',

        ['Region' => 'prelude'],

        ['Generic' => 'SYNOPSIS'],
        ['Generic' => 'DESCRIPTION'],
        ['Generic' => 'OVERVIEW'],
        ['Collect' => 'ATTRIBUTES' => {command => 'attr'}],
        ['Collect' => 'METHODS'    => {command => 'method'}],
        ['Collect' => 'FUNCTIONS'  => {command => 'func'}],

        'Leftovers',

        ['Region' => 'postlude'],

        'Bugs',
        'Authors',
        'Contributors',
        'Legal',

        ['Region' => 'footer'],
    );
}


sub mvp_bundle_config {
    my $self = shift || __PACKAGE__;

    return map { $self->_expand_config($_) } $self->configure;
}

sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub _expand_config {
    my $self = shift;
    my $spec = shift;

    my ($name, $package, $payload);

    if (!ref $spec) {
        ($name, $package, $payload) = ($spec, $spec, {});
    }
    elsif (@$spec == 1) {
        ($name, $package, $payload) = (@$spec[0,0], {});
    }
    elsif (@$spec == 2) {
        ($name, $package, $payload) = ref $spec->[1] ? @$spec[0,0,1] : (@$spec[1,0], {});
    }
    else {
        ($package, $name, $payload) = @$spec;
    }

    $name =~ s/^[@=-]//;
    $package = _exp($package);

    if ($package eq _exp('Region')) {
        $name = $spec->[1];
        $payload = {region_name => $spec->[1], %$payload};
    }

    $name = '@Author::CCM/' . $name if $package ne _exp('Generic') && $package ne _exp('Collect');

    return [$name => $package => $payload];
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Pod::Weaver::PluginBundle::Author::CCM - A plugin bundle for pod woven by CCM

=head1 VERSION

version 0.012

=head1 SYNOPSIS

    # In your weaver.ini file:
    [@Author::CCM]

    # In your dist.ini file:
    [PodWeaver]
    config_plugin = @Author::CCM

=head1 DESCRIPTION

You probably don't want to use this.

=head1 METHODS

=head2 configure

Returns the configuration in a form similar to what one might use with
L<Dist::Zilla::Role::PluginBundle::Easy/add_plugins>.

=head2 mvp_bundle_config

Required in order to be a plugin bundle.

=head1 SEE ALSO

=over 4

=item *

L<Pod::Weaver>

=item *

L<Pod::Weaver::PluginBundle::Author::ETHER>

=back

=head1 CREDITS

This module was heavily inspired by Karen Etheridge's config.

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website
L<https://github.com/chazmcgarvey/Dist-Zilla-PluginBundle-Author-CCM/issues>

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

Charles McGarvey <ccm@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Charles McGarvey.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.024.
use strict;
use warnings;

use 5.014;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "A plugin bundle for distributions built by CCM",
  "AUTHOR" => "Charles McGarvey <ccm\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Author-CCM",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.014",
  "NAME" => "Dist::Zilla::PluginBundle::Author::CCM",
  "PREREQ_PM" => {
    "Dist::Zilla" => "5.038",
    "Dist::Zilla::Plugin::Authority" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::BlockRelease" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::Encoding" => 0,
    "Dist::Zilla::Plugin::ExecDir" => 0,
    "Dist::Zilla::Plugin::Git::Check" => 0,
    "Dist::Zilla::Plugin::Git::Commit" => 0,
    "Dist::Zilla::Plugin::Git::CommitBuild" => 0,
    "Dist::Zilla::Plugin::Git::Contributors" => 0,
    "Dist::Zilla::Plugin::Git::GatherDir" => 0,
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::Git::Push" => 0,
    "Dist::Zilla::Plugin::Git::Tag" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => 0,
    "Dist::Zilla::Plugin::Keywords" => 0,
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker" => 0,
    "Dist::Zilla::Plugin::Manifest" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaNoIndex" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
    "Dist::Zilla::Plugin::MetaTests" => 0,
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::OverridePkgVersion" => 0,
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::Prereqs::AuthorDeps" => 0,
    "Dist::Zilla::Plugin::PruneCruft" => 0,
    "Dist::Zilla::Plugin::PruneFiles" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
    "Dist::Zilla::Plugin::RunExtraTests" => 0,
    "Dist::Zilla::Plugin::Test::CPAN::Changes" => 0,
    "Dist::Zilla::Plugin::Test::CleanNamespaces" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::EOL" => 0,
    "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
    "Dist::Zilla::Plugin::Test::NoTabs" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::Pod::No404s" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::Plugin::UploadToCPAN" => 0,
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
    "Dist::Zilla::Util" => 0,
    "Moose" => 0,
    "Perl::Version" => 0,
    "Pod::Elemental::Transformer::List" => 0,
    "Pod::Elemental::Transformer::Verbatim" => 0,
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::EnsurePod5" => 0,
    "Pod::Weaver::Plugin::H1Nester" => 0,
    "Pod::Weaver::Plugin::SingleEncoding" => 0,
    "Pod::Weaver::Plugin::Transformer" => 0,
    "Pod::Weaver::Section::Authors" => 0,
    "Pod::Weaver::Section::Bugs" => 0,
    "Pod::Weaver::Section::Collect" => 0,
    "Pod::Weaver::Section::Contributors" => 0,
    "Pod::Weaver::Section::Generic" => 0,
    "Pod::Weaver::Section::Legal" => 0,
    "Pod::Weaver::Section::Name" => 0,
    "Pod::Weaver::Section::Region" => 0,
    "Pod::Weaver::Section::Version" => 0,
    "Test::CPAN::Meta" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.012",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla" => "5.038",
  "Dist::Zilla::Plugin::Authority" => 0,
  "Dist::Zilla::Plugin::AutoPrereqs" => 0,
  "Dist::Zilla::Plugin::BlockRelease" => 0,
  "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
  "Dist::Zilla::Plugin::ConfirmRelease" => 0,
  "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
  "Dist::Zilla::Plugin::Encoding" => 0,
  "Dist::Zilla::Plugin::ExecDir" => 0,
  "Dist::Zilla::Plugin::Git::Check" => 0,
  "Dist::Zilla::Plugin::Git::Commit" => 0,
  "Dist::Zilla::Plugin::Git::CommitBuild" => 0,
  "Dist::Zilla::Plugin::Git::Contributors" => 0,
  "Dist::Zilla::Plugin::Git::GatherDir" => 0,
  "Dist::Zilla::Plugin::Git::NextVersion" => 0,
  "Dist::Zilla::Plugin::Git::Push" => 0,
  "Dist::Zilla::Plugin::Git::Tag" => 0,
  "Dist::Zilla::Plugin::GithubMeta" => 0,
  "Dist::Zilla::Plugin::Keywords" => 0,
  "Dist::Zilla::Plugin::License" => 0,
  "Dist::Zilla::Plugin::MakeMaker" => 0,
  "Dist::Zilla::Plugin::Manifest" => 0,
  "Dist::Zilla::Plugin::ManifestSkip" => 0,
  "Dist::Zilla::Plugin::MetaJSON" => 0,
  "Dist::Zilla::Plugin::MetaNoIndex" => 0,
  "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
  "Dist::Zilla::Plugin::MetaTests" => 0,
  "Dist::Zilla::Plugin::MetaYAML" => 0,
  "Dist::Zilla::Plugin::NextRelease" => 0,
  "Dist::Zilla::Plugin::OverridePkgVersion" => 0,
  "Dist::Zilla::Plugin::PodCoverageTests" => 0,
  "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::Prereqs::AuthorDeps" => 0,
  "Dist::Zilla::Plugin::PruneCruft" => 0,
  "Dist::Zilla::Plugin::PruneFiles" => 0,
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
  "Dist::Zilla::Plugin::RunExtraTests" => 0,
  "Dist::Zilla::Plugin::Test::CPAN::Changes" => 0,
  "Dist::Zilla::Plugin::Test::CleanNamespaces" => 0,
  "Dist::Zilla::Plugin::Test::Compile" => 0,
  "Dist::Zilla::Plugin::Test::EOL" => 0,
  "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
  "Dist::Zilla::Plugin::Test::NoTabs" => 0,
  "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
  "Dist::Zilla::Plugin::Test::Pod::No404s" => 0,
  "Dist::Zilla::Plugin::Test::Portability" => 0,
  "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
  "Dist::Zilla::Plugin::TestRelease" => 0,
  "Dist::Zilla::Plugin::UploadToCPAN" => 0,
  "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
  "Dist::Zilla::Util" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => 0,
  "Perl::Version" => 0,
  "Pod::Elemental::Transformer::List" => 0,
  "Pod::Elemental::Transformer::Verbatim" => 0,
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::EnsurePod5" => 0,
  "Pod::Weaver::Plugin::H1Nester" => 0,
  "Pod::Weaver::Plugin::SingleEncoding" => 0,
  "Pod::Weaver::Plugin::Transformer" => 0,
  "Pod::Weaver::Section::Authors" => 0,
  "Pod::Weaver::Section::Bugs" => 0,
  "Pod::Weaver::Section::Collect" => 0,
  "Pod::Weaver::Section::Contributors" => 0,
  "Pod::Weaver::Section::Generic" => 0,
  "Pod::Weaver::Section::Legal" => 0,
  "Pod::Weaver::Section::Name" => 0,
  "Pod::Weaver::Section::Region" => 0,
  "Pod::Weaver::Section::Version" => 0,
  "Test::CPAN::Meta" => 0,
  "Test::More" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
